### PR TITLE
fix: preserve unsigned image tag upon rebase

### PR DESCRIPTION
**Pull request by gerblesh**
this should hopefully rectify the rebasing woes. Just need to test it, hopefully this should remove the blockers needed to integrate with bluefin

### CODE REVIEW

File: files/etc/ublue-update.d/system/00-system-update.sh [hunk 1]
```diff
@@ -3,15 +3,27 @@
...
+
+        LOCAL_IMAGE_PREFIX=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $1}')
         LOCAL_IMAGE_REF_UNTAGGED=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $1":"$2":"$3}')
+        LOCAL_IMAGE_REF_TAG=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $4}')
+
         IMAGE_REF=$(jq -r '."image-ref"' < "$IMAGE_REF_FILE")
         IMAGE_TAG=$(jq -r '."image-tag"' < "$IMAGE_REF_FILE")
 
         if [[ "$LOCAL_IMAGE_REF_UNTAGGED" != "$IMAGE_REF" ]]; then
+            if [[ "$LOCAL_IMAGE_PREFIX" == "ostree-unverified-image" ]]; then
```
Comment (bobslept) [inline files/etc/ublue-update.d/system/00-system-update.sh:23]:
```suggestion
            if [[ "$LOCAL_IMAGE_PREFIX" == "ostree-unverified-registry" ]]; then
```

File: files/etc/ublue-update.d/system/00-system-update.sh [hunk 2]
```diff
@@ -3,15 +3,27 @@
...
+        STATUS_COMMAND="rpm-ostree status --pending-exit-77 -b --json"
+        LOCAL_IMAGE_REF=$($STATUS_COMMAND | jq -r '.deployments[0]["container-image-reference"]')
+
+        if ! $STATUS_COMMAND >/dev/null || [[ "$LOCAL_IMAGE_REF" == "null" ]]; then
+            # if jq failed to get the variable, or rpm-ostree had a nonzero exit code, return
             return
         fi
+
+        LOCAL_IMAGE_PREFIX=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $1}')
         LOCAL_IMAGE_REF_UNTAGGED=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $1":"$2":"$3}')
```
Comment (bobslept) [inline files/etc/ublue-update.d/system/00-system-update.sh:16]:
```suggestion
        LOCAL_IMAGE_REF_UNTAGGED=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $1":"$2}')
```

File: files/etc/ublue-update.d/system/00-system-update.sh [hunk 3]
```diff
@@ -3,15 +3,27 @@
...
+        LOCAL_IMAGE_REF_TAG=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $4}')
```
Comment (bobslept) [inline files/etc/ublue-update.d/system/00-system-update.sh:17]:
```suggestion
        LOCAL_IMAGE_REF_TAG=$(echo "$LOCAL_IMAGE_REF" | awk -F ":" '{print $3}')
```

**Review by bobslept — CHANGES_REQUESTED** [review]
The new code didn't fix it for me. I have tested the code with `set -ouex pipefail` and debugged it a little. I have made some changes and now it does work. I have no idea if these changes will break other scenario's.

Thas has been done on fresh `bluefin-dx:38` and `bluefin-dx:latest`

File: files/etc/ublue-update.d/system/00-system-update.sh (hunk 2 above)
Comment (bobslept) [reply files/etc/ublue-update.d/system/00-system-update.sh:16]: Old code will get you 
```
++ echo ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx:38
++ awk -F : '{print $1":"$2":"$3}'
+ LOCAL_IMAGE_REF_UNTAGGED=ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx:38
```

New code will get you 
```
++ echo ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx:38
++ awk -F : '{print $1":"$2}'
+ LOCAL_IMAGE_REF_UNTAGGED=ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx
```

File: files/etc/ublue-update.d/system/00-system-update.sh (hunk 1 above)
Comment (bobslept) [reply files/etc/ublue-update.d/system/00-system-update.sh:23]: Old code would give this
```
+ [[ ostree-unverified-registry == \o\s\t\r\e\e\-\u\n\v\e\r\i\f\i\e\d\-\i\m\a\g\e ]]
```
New code  will give this
```
+ [[ ostree-unverified-registry == \o\s\t\r\e\e\-\u\n\v\e\r\i\f\i\e\d\-\r\e\g\i\s\t\r\y ]]
```

File: files/etc/ublue-update.d/system/00-system-update.sh (hunk 3 above)
Comment (bobslept) [reply files/etc/ublue-update.d/system/00-system-update.sh:17]: Old code will get you
```
++ echo ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx:38
++ awk -F : '{print $4}'
+ LOCAL_IMAGE_REF_TAG=
```
New code will get you
```
++ echo ostree-unverified-registry:ghcr.io/ublue-os/bluefin-dx:38
++ awk -F : '{print $3}'
+ LOCAL_IMAGE_REF_TAG=38
```

File: files/etc/ublue-update.d/system/00-system-update.sh (hunk 2 above)
Comment (gerblesh) [reply files/etc/ublue-update.d/system/00-system-update.sh:16]: should account for the shorthand now

**Review by bobslept — APPROVED** [review]
Awesome. I've tested both `ostree-unverified-image:docker://` and `ostree-unverified-registry:` installations and it switched them both to a signed version. With the installaed tag in place. Perfect, thanks!

**Pull request merged**